core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java

```diff
@@ -0,0 +1,66 @@
+package io.quarkus.annotation.processor;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class Constants {
+    public static final String PARENT = "<<parent>>";
+    public static final String NO_DEFAULT = "<<no default>>";
+    public static final String HYPHENATED_ELEMENT_NAME = "<<hyphenated element name>>";
+
+    public static final Pattern JAVA_DOC_CODE_PATTERN = Pattern.compile("\\{@code (.*?)\\}");
```

Comment:
Here's what I sent to @gsmet before I left on PTO:

> The way it works is, the Quarkus extension
`quarkus-extension-processor` runs on each extension module which
contains build steps (obviously deployment modules only) or
configuration (both deployment and runtime).  It generates a file
`META-INF/quarkus-javadoc.properties` that contains key-value pairs
where the key is `package.name.ClassName.methodName` and the value is
the complete JavaDoc text including tags.
>
> The configuration classes can be found by tracing the path from
classes marked with `@ConfigRoot` annotations.  We do not presently
Jandex our extensions.  To find the config classes in use, you have to
iterate the classes named in `META-INF/quarkus-build-steps.list` and
then iterate all the constructor parameters, non-final fields, and
parameters of any method marked with the `@BuildStep` annotation and
locate any class named there that has `@ConfigRoot` on it.
>
> This is obviously not ideal and I'd like to simplify it.  I want to
move to putting a Jandex index on every extension deployment and
runtime module.  I also think we should be putting the config member
javadoc in the Jandex index just so we can have one place to look;
however this is a bit tricky to do in a way which isn't overly complex
or which doesn't include unwanted JavaDoc.  Then I'd like to locate
all configuration classes simply by examining the Jandex index, rather
than traversing outwards from the build steps.
>
> The correct approach for generating documentation for these things is
to start by parsing the JavaDoc, in two stages.  In the first stage
you expand inline doc tags and isolate the block tags into separate
units - this gives us the chance to introduce our own doc tags, if we
want.  In the second stage you parse them as HTML using something like
jsoup.  Once the JavaDoc is parsed into some kind of structure, you
can then render it in the desired format (plain text, adoc, md, HTML5,
etc.).
>
> Stuart bypassed a lot of this by just slamming it with regex
substitutions.  That only works up to a point though.
>
> Hope this helps.

The output is always going to be janky if we're going to be using regex to "parse" the docs.  If we want clean formatting, the parser needs semantic awareness of what it has parsed.  Would be nice to do this right.

Reply:
So this have moved a bit since the last implementation. I am now using a combination of `jsoup` and `javaparser` libs. WDYT?